tests/snippets/class.py

```diff
@@ -164,3 +164,25 @@ class T5(int):
 assert str(super(int, T5(5))) == "<super: <class 'int'>, <T5 object>>"
 
 #assert str(super(type, None)) == "<super: <class 'type'>, NULL>"
+
+a = 1
+class A:
+    a = 2
+    def b():
+        assert a == 1
+    b()
+    assert a == 2
+A.b()
+
+# TODO: uncomment once free vars/cells are working
```

Comment:
Worth a comment about how/why this is currently broken?  Without testing, I'm guessing the symbol table spots the ```a = 2```, and marks a as being nonlocal, but then the load_name still walks the incorrect scope list and gets ```a = 3```.

Reply:
Yep, exactly. I'll add a comment.